### ISSUE TITLE
Removing a hasOne relationship

### ISSUE THREAD

**Issue by danherd** [cloudcreativity/laravel-json-api#119]
This should work for removing an optional hasOne relationship:

PATCH /locums/id/relationships/region

```
{
    "data": null
}
```

but it doesn't:

Internal Server Error 500
Data member is not an object.
/vendor/cloudcreativity/json-api/src/Object/Document.php

Any way I can get this to work (I don't mind using a hacky workaround if needed)?

Cheers,

Dan

**Comment by danherd** [comment]
Ah, never mind - I've realised I can do it like this:

PATCH /locums/id

```
{
	"data": {
		"type": "locums",
		"id": "id",
		"relationships": {
			"region": {
			    "data": null
			}
		}
	}
}
```

Be good if I could also do it the other way though.

**Comment by lindyhopchris** [comment]
Yeah, thanks for reporting. I'm working on the relationship stuff at the moment so will fix and add tests for this as part of that work.

**Comment by lindyhopchris** [comment]
Note to self - need to add tests for a has-one Eloquent relationship. The JSON API has-one relation on `1.x-dev` won't work with it because of the use of `associate()` and `dissociate()` in the `update` method.

**Comment by lindyhopchris** [comment]
This is now implemented on the 1.0 branch.
